src/cocotb/regression.py

```diff
@@ -90,24 +91,61 @@
         assert False, "pytest.raises doesn't raise an exception when it fails"
 
 
-class _Test:
-    _id_count = 0  # used by the RegressionManager to sort tests in definition order
+class Test:
+    """A cocotb test in a regression.
+
+    Args:
+        func:
+            The test function.
+
+        name:
+            The name of the test function.
+            Defaults to ``func.__qualname__`` (the dotted path to the test function in the module) if not given.
+
+        module:
+            The name of the module containing the test function.
+            Defaults to ``func.__module__`` (the name of the module containing the test function) if not given.
+
+        doc:
+            The docstring for the test.
+            Defaults to ``func.__doc__`` (the docstring of the test function) if not given.
+
+        timeout_time:
+            Simulation time duration before timeout occurs.
+
+        timeout_unit:
+            Units of ``timeout_time``, accepts any units that :class:`~cocotb.triggers.Timer` does.
+
+        expect_fail:
+            If ``True`` and the test fails a functional check via an ``assert`` statement, :class:`pytest.raises`,
+            :class:`pytest.warns`, or :class:`pytest.deprecated_call`, the test is considered to have passed.
+            If ``True`` and the test passes successfully, the test is considered to have failed.
+
+        expect_error:
+            Mark the result as a pass only if one of the given exception types is raised in the test.
+
+        skip:
+            Don't execute this test as part of the regression.
+            The test can still be run manually by setting :make:var:`TESTCASE`.
+
+        stage:
+            Order tests logically into stages, where multiple tests can share a stage.
```

Comment:
I have never intuitively gotten what this does TBH...

Reply:
IMO it should be removed... but maybe someone is using it and we can't really provide an alternative outside of "lexically reorder your tests in the file", which is poopy. 

It's there to mark that some tests should run before/after other tests based on the stage number. So all stage=0 tests run, then all stage=1, then all stage=2, etc.

I'm fine with taking the opportunity to deprecate this without alternative, but will try to come up with a better docstring.